content/en/blog/2025-12-12-flux-aio-kubernetes-mtls-and-the-chicken-and-egg-problem/index.md

```
@@ -0,0 +1,132 @@
+---
+title: "Flux-aio, Kubernetes mTLS and the Chicken and Egg Problem"
+slug: flux-aio-kubernetes-mtls-and-the-chicken-and-egg-problem
+date: 2025-12-12
+author: "Andrei Kvapil"
+description: "How we solved the chicken-and-egg problem of deploying CNI and kube-proxy through Flux while ensuring Flux itself works without CNI and kube-proxy, using Kubernetes API routing and mTLS certificates."
+---
+
+![](chicken-and-egg-problem.png)
```

markdownlint-cli2 (0.18.1)

Images should have alternate text (alt text)

(MD045, no-alt-text)